### DISCUSSION TITLE
Is it possible to prepare a tweet with more than 280 characters?

### DISCUSSION THREAD

**Discussion by arnonrdp** [tweepy/tweepy#1749] [Questions] [answered]
Currently, I've been handling large tweets using the `wrap()` function, but I'm wondering if Tweepy doesn't already have some method to do this.

I created these two functions to tweet texts longer than 280 characters:

```py
def prepare_tweet(tweet_list, api):
    for tweet in tweet_list:
        tweet_encoded = tweet.encode("utf-8", "ignore")
        tweet_decoded = tweet_encoded.decode()
        print('\n' + tweet_decoded + '\n')
        each_tweet = wrap(tweet_decoded, 280, break_long_words=False)
        post_tweet(api, each_tweet)


def post_tweet(api, each_tweet):
    original_tweet = []
    for i, chunk in zip(range(len(each_tweet)), each_tweet):
        original_tweet.extend([chunk])
        if i == 0:
            original_tweet[i] = api.update_status(chunk)
        else:
            original_tweet[i] = api.update_status(chunk,
                                                  in_reply_to_status_id=original_tweet[i - 1].id,
                                                  auto_populate_reply_metadata=True)
```

`tweet_list` is a list of about 10 items/texts, all longer than 280 characters.
Some of this text is not in `utf-8`, so `.encode` and `.decode` it.

So basically my question is: is there any method that allows me to post a text longer than 280 characters and take care of slicing the tweets? (whereas the continuation of a text must be a reply to itself).

**Comment by Harmon758** [comment, marked as answer]
No, Tweepy doesn't currently support this with a single method. The relevant issue for this would be #1376.
